x/feegrant/keeper/keeper.go

```diff
@@ -52,8 +53,12 @@ func (k Keeper) GrantAllowance(ctx sdk.Context, granter, grantee sdk.AccAddress,
 	key := feegrant.FeeAllowanceKey(granter, grantee)
 
 	var oldExp *time.Time
-	existingGrant, err := k.getGrant(ctx, grantee, granter)
-	if err != nil && existingGrant != nil && existingGrant.GetAllowance() != nil {
+	existingGrant, err := k.getGrant(ctx, granter, grantee)
+	if err != nil && !strings.Contains(err.Error(), sdkerrors.ErrUnauthorized.Error()) {
```

Comment:
what happened to the comment here? We should not use `strings.Contains`.

Reply:
unable to match errors. Tried using `errors.Is` or `errors.As` too.